libsquoosh/src/index.ts

```diff
@@ -5,10 +5,18 @@ import { promises as fsp } from 'fs';
 import { codecs as encoders, preprocessors } from './codecs.js';
 import WorkerPool from './worker_pool.js';
 import { autoOptimize } from './auto-optimizer.js';
+import type ImageData from './image_data';
 
 export { ImagePool, encoders, preprocessors };
+type EncoderKey = keyof typeof encoders;
+type PreprocessorKey = keyof typeof preprocessors;
+type FileLike = Buffer | ArrayBuffer | string | { buffer: Buffer };
```

Comment:
I think we need `ArrayBufferView` in here as well (ArrayBuffer.isView is a type guard for `ArrayBufferView`. Then, we can remove `{ buffer: Buffer }` typing from FileLike since ArrayBufferView already contains `buffer` property. WDYT?

Reply:
ArrayBufferView didn't work with the existing code

![image](https://user-images.githubusercontent.com/229881/127586377-a416cf4f-055d-4ec8-95f2-a21907115ea6.png)

Reply:
Yes, that's strange and I didn't understand correctly why. However, I think it would be better to use `ArrayBufferView` in type definition and do a type assertion to `any` for `file` in the place with a TODO to figure it out why we need that. (i.e. maybe a type guard in the if clauses also checks for Buffer and we might not need `file instanceof Buffer` check in there) WDYT?